python/pyarrow/tests/test_scalars.py

```diff
@@ -171,3 +171,37 @@ def test_dictionary(self):
                                            categorical.categories)
         for i, c in enumerate(values):
             assert v[i].as_py() == c
+
+    def test_int_hash(self):
+        # ARROW-640
+        arr = pa.array([1, 1, 2, 1])
+        arr2 = pa.array([1, 1, 2, 1])
+        assert arr[0].__hash__() == arr2[0].__hash__()
+
+    def test_int_eq(self):
```

Comment:
No need to add this test, it is implicitly tested for in `test_int64` above.

Reply:
Removed. Thank you for letting me know that.